splunk_logger.go

```diff
@@ -517,6 +524,73 @@ func (l *splunkLogger) queueMessageAsync(message *splunkMessage) error {
 	return nil
 }
 
+func telemetry(info logger.Info, l *splunkLogger, sourceType string, splunkFormat string) {
+
+	//Send weekly
+	timer := time.NewTicker(600000000000000)
+	messageArray := []*splunkMessage{}
+	timestamp := strconv.FormatInt(time.Now().UTC().UnixNano()/int64(time.Second), 10)
+
+	type telemetryEvent struct {
+		Component string `json:"component"`
+		Type      string `json:"type"`
+		Data      struct {
+			JsonLogs                bool   `json:"jsonLogs"`
+			PartialMsgBufferMaximum int    `json:"partialMsgBufferMaximum"`
+			PostMessagesBatchSize   int    `json:"postMessagesBatchSize"`
+			PostMessagesFrequency   int    `json:"postMessagesFrequency"`
+			SplunkFormat            string `json:"splunkFormat"`
+			StreamChannelSize       int    `json:"streamChannelSize"`
+			Sourcetype              string `json:"sourcetype"`
+		} `json:"data"`
+		OptInRequired int64 `json:"optInRequired"`
+	}
+
+	telem := &telemetryEvent{}
+	telem.Component = "app.connect.docker"
+	telem.Type = "event"
+	telem.OptInRequired = 2
+	telem.Data.Sourcetype = sourceType
+	telem.Data.SplunkFormat = splunkFormat
+	telem.Data.PostMessagesFrequency = int(getAdvancedOptionDuration(envVarPostMessagesFrequency, defaultPostMessagesFrequency))
+	telem.Data.PostMessagesBatchSize = getAdvancedOptionInt(envVarPostMessagesBatchSize, defaultPostMessagesBatchSize)
+	telem.Data.StreamChannelSize = getAdvancedOptionInt(envVarStreamChannelSize, defaultStreamChannelSize)
+	telem.Data.PartialMsgBufferMaximum = getAdvancedOptionInt(envVarBufferMaximum, defaultBufferMaximum)
+	telem.Data.JsonLogs, _ = strconv.ParseBool(info.Config[envVarJSONLogs])
+
+	var telemMessage = &splunkMessage{
+		Host:       "telemetry",
+		Source:     "telemetry",
```

Comment:
Are these right? 
Host should be system host?
source should be container id?

How can we differentiate at the telemetry end if a customer has 10 containers with the logger running sending to the same Splunk Cluster? Would source and host be the place to do this.

Reply:
Host and source are not needed for these events.